### DISCUSSION TITLE
How to forward data string from capacitor plugin to MainActivity android

### DISCUSSION THREAD

**Discussion by lebaphi** [ionic-team/capacitor#5361] [Q&A] [answered]
I wrote a simple capacitor plugin (SendJSON) that send the json string from the angular side to android side. I can received this string in the plugin side but I don't know how to forward this string to MainActivity and print out it. I've tried to use Intent but it does not work, I can not find the MainActivity.class in the plugin class.

MyAndroidPluginClass.java

```
@CapacitorPlugin(name = "JSDataBridge")
public class JSDataBridgePlugin extends Plugin {

    private JSDataBridge implementation = new JSDataBridge();

    @PluginMethod
    public void sendJSON(PluginCall call) {
        String value = call.getString("value");
        JSObject ret = new JSObject();
        ret.put("value", implementation.sendJSON(value));
        call.resolve(ret);
    }
}
```

I want to print the value here

MainActivity.java
```
public class MainActivity extends BridgeActivity {
  @Override
  public void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    // print the value that receive from the plugin here
  }
}
```
So how can I archive this thing?

**Comment by lebaphi** [comment, marked as answer]
Closed, I found a solution to do this.

**Reply by luckhoi56** [reply]
@lebaphi 
I am writing a custom plugin and having the same problem. Do you mind sharing your solution? Thanks

**Reply by lebaphi** [reply]
> @lebaphi I am writing a custom plugin and having the same problem. Do you mind sharing your solution? Thanks

Hi @luckhoi56 , please take a look at this article https://stackoverflow.com/questions/3947641/android-equivalent-to-nsnotificationcenter.
